x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichPlugin.java

```diff
@@ -61,6 +61,8 @@
 
     static final Setting<Integer> ENRICH_FETCH_SIZE_SETTING =
         Setting.intSetting("index.xpack.enrich.fetch_size", 10000, 1, 1000000, Setting.Property.NodeScope);
+    static final Setting<Integer> ENRICH_POLICY_MAX_CONCURRENT_RUNS =
+        Setting.intSetting("index.xpack.enrich.policy.max_concurrent_runs", 50, 1, Setting.Property.NodeScope);
```

Comment:
`50` default seems abit aggressive ? How about ` EsExecutors.numberOfProcessors(settings)` ?

Reply:
I think something like 50 is ok? This is to avoid policy execution abuse. The heavy bit is reindexing and force merging, but I don't think these kind of operations are bound to the number of processors on a node.

Reply:
I suggested processors as a proxy to how large of a total cluster/processing could be...but I agree it is not a good proxy. Are there concerns about 50 concurrent re-indexes in a small cluster ?

Reply:
In small or under sized clusters, the reindex & force merge would not run in parallel and these operations could get queued up in the search, write and force merge threadpool. But I don't think that is a real issue, policy execution would just be slower.